### PR TITLE
Fixes #429: Customization Menu for media walls

### LINKED ISSUE #429: Adding customization menu for media walls.
Currently, we have a basic prototype ready. Next thing to do is to make customization menu to help user personalize the media wall. In the first step, we would adding option to change just the search term. 

screen shot 2017-06-27 at 6 45 30 am

Please suggest @hemantjadon @mariobehling @sudheesh001 .

### PR DESCRIPTION
**Pull request by Achint08**
**Changes proposed in this pull request**
- Customization Menu for media walls

**Screenshots (if appropriate)** 

screen shot 2017-07-22 at 1 31 05 pm

**Closes #429 **

### CODE REVIEW

**Comment by Achint08** [comment]
Open for review @mariobehling @hemantjadon @djmgit @kavithaenair @vibhcool @singhpratyush @SKrPl .

File: src/app/media-wall/media-wall-customization/media-wall-customization.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+@Component({
+	selector: 'media-wall-custom-card',
+	template: ''
+})
+class MediaWallCustomCardStubComponent {
+	@Input() mediaWallCard;
+}
+
+
```
Comment (vibhcool) [inline src/app/media-wall/media-wall-customization/media-wall-customization.component.spec.ts]: extra line

File: src/app/media-wall/media-wall-customization/media-wall-customization.component.scss [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+			padding: 10px 25px;
+			font-size: 16px;
+			border-radius: 5px;
+			text-decoration: none;
+			margin-top: 10px;
+			cursor: pointer;
+	}
+}
+
+
```
Comment (vibhcool) [inline src/app/media-wall/media-wall-customization/media-wall-customization.component.scss]: lines at end

File: src/app/media-wall/media-wall-custom-background/media-wall-custom-background.component.scss [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+			border-radius: 5px;
+			text-decoration: none;
+			margin-top: 10px;
+			cursor: pointer;
+
+			.material-icons {
+				position: relative;
+				top: 4px;
+				right: 5px;
+			}
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall-custom-background/media-wall-custom-background.component.scss]: Indentation of this block

File: src/app/media-wall/media-wall-custom-background/media-wall-custom-background.component.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	labelPosition="before"
+	ariaLabel="Transparent Backgorund"
+	(change)="transparentBackground($event)">
+		Transparent Background
+	</md-slide-toggle>
+	</div>
+	<div class="color">
+		<label for="color">Background Color</label>
+			<input name="color"
+			[cpOutputFormat]="'hex'"
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall-custom-background/media-wall-custom-background.component.html]: give one tab indent to this

File: src/app/media-wall/media-wall-custom-background/media-wall-custom-background.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+			imports: [,
+				MdSlideToggleModule.forRoot(),
+				MaterialModule
+			],
+			declarations: [
+				MediaWallCustomBackgroundComponent
+			]
+		});
+
+	it('should create an instance', () => {
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall-custom-background/media-wall-custom-background.component.spec.ts]: Indentation error, one less indent

File: src/app/media-wall/media-wall-custom-card/media-wall-custom-card.component.html [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+	labelPosition="before"
+	ariaLabel="Transparent Backgorund"
+	(change)="transparentBackground($event)">
+		Transparent Background
+	</md-slide-toggle>
+	</div>
+	<div class="color">
+		<label for="color">Background Color</label>
+			<input name="color"
+			[cpOutputFormat]="'hex'"
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall-custom-card/media-wall-custom-card.component.html]: Indentation

File: src/app/media-wall/media-wall-custom-card/media-wall-custom-card.component.html [hunk 7]
```diff
@@ -0,0 +1,63 @@
+<div class="tab">
+<h1>Card Customization</h1>
+<h3>Card Background</h3>
+<div class="full-width">
+	<div class="slider">
+	<md-slide-toggle
+	labelPosition="before"
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall-custom-card/media-wall-custom-card.component.html]: indentation

File: src/app/media-wall/media-wall-custom-card/media-wall-custom-card.component.html [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+			[value]="customCard.backgroundColor"
+			(colorPickerChange)="changeBackgroundColor($event)" />
+	</div>
+</div>
+<h3>Card Font</h3>
+<div class="full-width">
+	<div class="color">
+		<label for="color">Font Color</label>
+			<input name="color"
+			[cpOutputFormat]="'hex'"
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall-custom-card/media-wall-custom-card.component.html]: Indentation

File: src/app/media-wall/media-wall-custom-card/media-wall-custom-card.component.html [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+			[value]="customCard.fontColor"
+			(colorPickerChange)="changeFontColor($event)" />
+	</div>
+</div>
+<h3>Card Accent</h3>
+<div class="full-width">
+	<div class="color">
+		<label for="color">Accent Color</label>
+			<input name="color"
+			[cpOutputFormat]="'hex'"
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall-custom-card/media-wall-custom-card.component.html]: Indentation

File: src/app/media-wall/media-wall-custom-card/media-wall-custom-card.component.scss [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+		}
+	}
+}
+
+span.apply-button {
+	margin-right: auto;
+	margin-left: auto;
+
+	button {
+			border: none;
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall-custom-card/media-wall-custom-card.component.scss]: indentation of this block

File: src/app/media-wall/media-wall-custom-card/media-wall-custom-card.component.spec.ts [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+			imports: [,
+				MdSlideToggleModule.forRoot(),
+				MaterialModule
+			],
+			declarations: [
+				MediaWallCustomCardComponent
+			]
+		});
+
+	it('should create an instance', () => {
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall-custom-card/media-wall-custom-card.component.spec.ts]: Indentation of this function

File: src/app/media-wall/media-wall-customization/media-wall-customization.component.spec.ts (hunk 1 above)
Comment (Achint08) [inline src/app/media-wall/media-wall-customization/media-wall-customization.component.spec.ts]: For user readibility.

**Comment by Achint08** [comment]
@hemantjadon @vibhcool Please review now.
Also updated the test link 
https://achint08.github.io/loklak_search/

**Comment by vibhcool** [comment]
could scroller be removed?
browser: firefox 54
screenshot from 2017-07-24 03-00-34

**Comment by Achint08** [comment]
@vibhcool 

> I checked the checkbox for transparent background and applied.
When I reopen, it shows it unchecked
It appears to be a bug :P

Done! now👍 

And regarding the scroller, I added it keeping in mind the smaller screens. I need to make adjustments for the responsiveness of this dialog box. I will look into this later.

**Comment by mariobehling** [comment]
* Screen is partly off screen. App apparently has not been tested on different resolutions.
screenshot from 2017-07-25 10-48-08
* "Full Screen" box is also off screen partly
* "loklak" and central search term here "fossasia" is not in one line
screenshot from 2017-07-25 10-48-38

**Comment by Achint08** [comment]
@mariobehling Improving it now.
I think `Md-dialog` is not detecting the screen `%` correctly.That's why it is off-screen.

**Comment by Achint08** [comment]
@mariobehling Done! Please check out now. 
Just for the smaller resolution, since we can't show inline-color-picker(due to less space). I need to make it fixed. I will add it in subsequent PRs.

**Comment by mariobehling** [comment]
Ok, thanks.

**Pull request merged**